pkg/sources/transformer/builtin/event_time/event_time_extractor.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventtime
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	functionsdk "github.com/numaproj/numaflow-go/pkg/function"
+
+	"github.com/numaproj/numaflow/pkg/shared/expr"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+)
+
+type eventTimeExtractor struct {
+	// expression is used to extract the string representation of the event time from message payload.
+	// e.g. `json(payload).metadata.time`
+	expression string
+	// format specifies the layout of extracted time string.
+	// With format, we use the time.Parse function to translate the event time string representation to time.Time object.
+	format string
+}
+
+func New(args map[string]string) (functionsdk.MapTFunc, error) {
+	expr, existing := args["expression"]
+	if !existing {
+		return nil, fmt.Errorf("missing \"expression\"")
```

Comment:
```suggestion
		return nil, fmt.Errorf(`missing "expression"`)
```

NIT: use grave quotes